lib/controllers/root-controller.js

```diff
@@ -362,13 +367,15 @@ export default class RootController extends React.Component {
   }
 
   @autobind
-  async initializeRepo() {
+  async initializeRepo(initDialogPath) {
     if (this.props.activeWorkingDirectory) {
       await this.acceptInit(this.props.activeWorkingDirectory);
-      return;
+      return null;
```

Comment:
:point_up: I think we should ignore the active working directory if an explicit path to initialize is given:

```js
if (!initDialogPath || this.props.activeWorkingDirectory) {
  // ...
}
```

Otherwise I worry that a caller could:

1. Generate the path to initialize
2. Yield the event loop, and something else triggers a `WorkdirContext` change
3. Call `initializeRepo` with the desired path
4. `initializeRepo` will silently init the active working directory instead!